cmd/lotus-shed/f3.go

```
@@ -61,6 +74,146 @@ func loadF3IDList(path string) ([]gpbft.ActorID, error) {
+		err = ethReturn.UnmarshalCBOR(bytes.NewReader(msgRes.MsgRct.Return))
+		if err != nil {
+			return fmt.Errorf("could not decode return value: %w", err)
+		}
+		fmt.Printf("Raw data: %X\n", ethReturn)
+		slot, retBytes := []byte{}, []byte(ethReturn)
+		// 3*32 because there should be 3 slots minimum
+		if len(retBytes) < 3*32 {
+			return fmt.Errorf("no activation infromation")
+		}
+
+		// split off first slot
+		slot, retBytes = retBytes[:32], retBytes[32:]
+		// it is uint64 so we want the last 8 bytes
+		slot = slot[24:32]
+		activationEpoch := binary.BigEndian.Uint64(slot)
+		_ = activationEpoch
+
+		slot, retBytes = retBytes[:32], retBytes[32:]
+		for i := 0; i < 31; i++ {
+			if slot[i] != 0 {
+				return fmt.Errorf("wrong value for offest (padding): slot[%d] = 0x%x != 0x00", i, slot[i])
+			}
+		}
+		if slot[31] != 0x40 {
+			return fmt.Errorf("wrong value for offest : slot[31] = 0x%x != 0x40", slot[31])
+		}
+		slot, retBytes = retBytes[:32], retBytes[32:]
+		slot = slot[24:32]
+		pLen := binary.BigEndian.Uint64(slot)
+		if pLen > 4<<10 {
+			return fmt.Errorf("too long declared payload: %d > %d", pLen, 4<<10)
+		}
+		payloadLength := int(pLen)
+
+		if payloadLength > len(retBytes) {
+			return fmt.Errorf("not enough remaining bytes: %d > %d", payloadLength, retBytes)
+		}
+
+		if activationEpoch == math.MaxUint64 || payloadLength == 0 {
+			fmt.Printf("no active activation")
+		} else {
+			compressedManifest := retBytes[:payloadLength]
+			reader := io.LimitReader(flate.NewReader(bytes.NewReader(compressedManifest)), 1<<20)
+			var m manifest.Manifest
+			err = json.NewDecoder(reader).Decode(&m)
+			if err != nil {
+				return fmt.Errorf("got error while decoding manifest: %w", err)
+			}
+
+			if m.BootstrapEpoch < 0 || uint64(m.BootstrapEpoch) != activationEpoch {
+				return fmt.Errorf("bootstrap epoch does not match: %d != %d", m.BootstrapEpoch, activationEpoch)
+			}
+			io.Copy(os.Stdout, flate.NewReader(bytes.NewReader(compressedManifest)))
```

Explicitly ignore the error?
```suggestion
			_, _ = io.Copy(os.Stdout, flate.NewReader(bytes.NewReader(compressedManifest)))
```